infrastructure/godisk/disk_service.go

```diff
@@ -0,0 +1,93 @@
+package godisk
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/diskfs/go-diskfs"
+	"github.com/diskfs/go-diskfs/disk"
+	"github.com/diskfs/go-diskfs/filesystem"
+	units "github.com/docker/go-units"
+	"github.com/spf13/afero"
+
+	"github.com/weaveworks-liquidmetal/flintlock/core/ports"
+)
+
+func New(fs afero.Fs) ports.DiskService {
+	return &diskService{
+		fs: fs,
+	}
+}
+
+type diskService struct {
+	fs afero.Fs
+}
+
+// Create will create a new disk.
+func (s *diskService) Create(ctx context.Context, input ports.DiskCreateInput) error {
+	if input.Path == "" {
+		return errPathRequired
+	}
+	if input.Size == "" {
+		return errSizeRequired
+	}
+
+	if _, err := s.fs.Stat(input.Path); err == nil {
+		if removeErr := s.fs.Remove(input.Path); removeErr != nil {
+			return fmt.Errorf("removing disk %s: %w", input.Path, removeErr)
+		}
+	} else if !errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("checking if disk exists: %w", err)
+	}
+
+	diskSize, err := units.FromHumanSize(input.Size)
+	if err != nil {
+		return fmt.Errorf("converting disk size %s: %w", input.Size, err)
+	}
+
+	createdDisk, err := diskfs.Create(input.Path, int64(diskSize), diskfs.Raw)
+	if err != nil {
+		return fmt.Errorf("creating disk %s: %w", input.Path, err)
+	}
+
+	createdDisk.LogicalBlocksize = 512
```

Comment:
can we have that in a const?

Reply:
Yep, that would've been a const to start with :+1: